src/analysis/processing/qgsalgorithmrasterize.cpp

```diff
@@ -150,6 +151,72 @@ QVariantMap QgsRasterizeAlgorithm::processAlgorithm( const QVariantMap &paramete
   int height { yTileCount * tileSize };
   int nBands { transparent ? 4 : 3 };
 
+  int64_t totalTiles = 0;
+  for ( auto &layer : std::as_const( mMapLayers ) )
+  {
+    if ( QgsRasterLayer *rasterLayer = qobject_cast<QgsRasterLayer *>( ( layer.get() ) ) )
+    {
+      if ( const QgsProviderMetadata *metadata = rasterLayer->providerMetadata() )
+      {
+        QVariantMap details = metadata->decodeUri( rasterLayer->source() );
+        if ( details.contains( QStringLiteral( "url" ) ) && details[QStringLiteral( "url" )].toString().startsWith( QStringLiteral( "https://tile.openstreetmap.org" ), Qt::CaseInsensitive ) )
```

Comment:
I'd suggest to pass the url to a `QUrl`, then compare its `host()` part to `tile.openstreetmap.org`.
This will avoid some very easy ways to circumvent the check, like passing a blank user/pass or urlencoding the dots.